tests/functional/src/main/java/org/mule/functional/junit4/TestLegacyMessageBuilder.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.junit4;
+
+import static org.mule.functional.junit4.TestLegacyMessageUtils.LEGACY_MESSAGE_API_ERROR;
+import org.mule.runtime.api.message.Message;
+import org.mule.runtime.api.metadata.MediaType;
+import org.mule.runtime.core.api.message.ExceptionPayload;
+
+import java.io.Serializable;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Builds messages supporting the legacy message API
+ *
+ * @deprecated tests should not access properties, attachments or exception payload using the old API.
+ */
+@Deprecated
+public class TestLegacyMessageBuilder implements Message.Builder {
+
+  private Message.Builder builder;
+
+  public TestLegacyMessageBuilder() {}
```

Comment:
I can't create a builder without setting the payload, as Message.builder() returns a PayloadBuilder. Do you think that doing Message.builder().nullPayload() is OK?